pkg/controller/map/storage/validation.go

```diff
@@ -33,21 +45,135 @@ const (
 //
 // Validate the mp resource.
 func (r *Reconciler) validate(mp *api.StorageMap) error {
-	provider := validation.ProviderPair{Client: r}
-	conditions, err := provider.Validate(mp.Spec.Provider)
+	pv := validation.ProviderPair{Client: r}
+	conditions, err := pv.Validate(mp.Spec.Provider)
 	if err != nil {
 		return err
 	}
 	mp.Status.UpdateConditions(conditions)
 	if mp.Status.HasCondition(validation.SourceProviderNotReady) {
 		return nil
 	}
-	storage := validation.StoragePair{Client: r, Provider: provider.Referenced}
-	conditions, err = storage.Validate(mp.Spec.Map)
+	mp.Referenced.Provider.Source = pv.Referenced.Source
+	mp.Referenced.Provider.Destination = pv.Referenced.Destination
+	err = r.validateSource(mp)
+	if err != nil {
+		return err
+	}
+	err = r.validateDestination(mp)
 	if err != nil {
 		return err
 	}
-	mp.Status.UpdateConditions(conditions)
 
 	return nil
 }
+
+//
+// Validate source refs.
+func (r *Reconciler) validateSource(mp *api.StorageMap) (err error) {
+	provider := mp.Referenced.Provider.Source
+	inventory, err := web.NewClient(provider)
+	if err != nil {
+		return
+	}
+	notValid := []string{}
+	ambiguous := []string{}
+	list := mp.Spec.Map
+	for i := range list {
+		ref := &list[i].Source
+		if ref.NotSet() {
+			mp.Status.SetCondition(libcnd.Condition{
+				Type:     DestinationStorageNotValid,
+				Status:   True,
+				Reason:   NotSet,
+				Category: Critical,
+				Message:  "Destination network: either `ID` or `Name` required.",
```

Comment:
I think the condition here is supposed to be `SourceStorageNotValid`, and the message should be `"Source storage: ..."`

Reply:
Fixed.